src/Kiota.Builder/Writers/Python/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Writers.Python;
+public class CodeClassDeclarationWriter : BaseElementWriter<ClassDeclaration, PythonConventionService>
+{
+    private readonly PythonRelativeImportManager _relativeImportManager;
+    public CodeClassDeclarationWriter(PythonConventionService conventionService, string clientNamespaceName) : base(conventionService){
+        _relativeImportManager = new PythonRelativeImportManager(clientNamespaceName, '.');
+    }
+    public override void WriteCodeElement(ClassDeclaration codeElement, LanguageWriter writer)
+    {
+        if(codeElement == null) throw new ArgumentNullException(nameof(codeElement));
+        if(writer == null) throw new ArgumentNullException(nameof(writer));
+        var parentNamespace = codeElement.GetImmediateParentOfType<CodeNamespace>();
+        WriteExternalImports(codeElement, writer); // external imports before internal imports
+        WriteInternalImports(codeElement, writer, parentNamespace);
+        
+        var inheritSymbol = conventions.GetTypeString(codeElement.Inherits, codeElement);
+        var abcClass = !codeElement.Implements.Any() ? string.Empty : $"{codeElement.Implements.Select(x => x.Name.ToFirstCharacterUpperCase()).Aggregate((x,y) => x + ", " + y)}";
+        var derivation = inheritSymbol == null ? abcClass : $"{inheritSymbol}";
+        if(codeElement.Parent?.Parent is CodeClass){
+            writer.WriteLine($"@dataclass");
+        }
+        writer.WriteLine($"class {codeElement.Name.ToFirstCharacterUpperCase()}({derivation}):");
+        writer.IncreaseIndent();
+        conventions.WriteShortDescription((codeElement.Parent as CodeClass)?.Description, writer);
+    }
+    
+    private static void WriteExternalImports(ClassDeclaration codeElement, LanguageWriter writer) {
+        var externalImportSymbolsAndPaths = codeElement.Usings
+                                                        .Where(x => x.IsExternal)
+                                                        .Select(x => (x.Name, string.Empty, x.Declaration?.Name))
+                                                        .GroupBy(x => x.Item3)
+                                                        .OrderBy(x => x.Key);
+        if(externalImportSymbolsAndPaths.Any()){
+            foreach (var codeUsing in externalImportSymbolsAndPaths)
+                if (!string.IsNullOrWhiteSpace(codeUsing.Key))
```

Comment:
collapse the if into a where in the pervious linq statements

Reply:
Thanks. Updated in https://github.com/microsoft/kiota/pull/1200/commits/cc585b8a61fbdb38b68c5e09d9ed56a06f554f11